examples/api/lib/painting/image_provider/image_provider.0.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+@immutable
+class CustomNetworkImage extends ImageProvider<Uri> {
+  const CustomNetworkImage(this.url);
+
+  final String url;
+
+  @override
+  Future<Uri> obtainKey(ImageConfiguration configuration) {
+    final Uri result = Uri.parse(url).replace(
+      queryParameters: <String, String>{
+        'dpr': '${configuration.devicePixelRatio}',
+        'locale': '${configuration.locale?.toLanguageTag()}',
+        if (configuration.platform == null)
+          'platform': null.toString(),
+        if (configuration.platform != null)
+          'platform': configuration.platform!.toString().substring('$TargetPlatform'.length + 1),
```

Comment:
Isn't `configuration.platform` an enum so this could be simplified to:

```suggestion
        'platform': configuration.platform?.name ?? null.toString(),
```

Reply:
oh I always forget about EnumName because it's not in the API docs for Enum

Reply:
Right! Send a PR to add it? ;)

Reply:
it simplified even more than you suggested, heh

Reply:
Beautiful!

Reply:
Submitted https://github.com/dart-lang/sdk/pull/53164 for the Enum doc issue.